### ISSUE TITLE
Undamaged Units acquire internal damage when loading campaign

### ISSUE THREAD

**Issue by crazys1** [MegaMek/mekhq#146]
MekHQ 0.3.32

After saving my campaign with all of my units being fully repaired, when I load the campaign again, over half of my 15 'Mechs acquire internal damage. In most cases, it is the complete loss of internal structure points in one or more locations, and sometimes it includes engine, gyro, jump jet, or actuator critical hits. If I repair the units with "GM Mode > Restore Unit", or by using "GM Mode > Edit Damage" and removing all damage, and then save the campaign again, loading the campaign results in identical damage being re-applied to all of the 'Mechs.

My campaign includes 6 aerospace fighters and a dropship, and so far these do not acquire this mystery damage. It seems to affect only the 'Mechs.

I attached the MekHQ log file as well as the campaign file.

To reproduce the bug:

1. Load the campaign file.
2. Go to the Hangar. Notice that several units are damaged internally.
3. Select all of the units.
4. Right click and choose "GM Mode > Restore Unit".
5. Save the campaign.
6. Reload the campaign.
7. Go to the Hangar. Notice the same units are damaged in exactly the same way.

[Nine-Tailed Foxes31450129.zip](https://github.com/MegaMek/mekhq/files/451323/Nine-Tailed.Foxes31450129.zip)

[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/451319/mekhqlog.txt)

**Comment by crazys1** [comment]
After making progress in the campaign, now at least one aerospace fighter has this issue, too. In addition, the "GM Mode > Restore Unit" feature does not repair the mystery damage for the ASF. I must manually remove the damage with "GM Mode > Edit Damage".

I attached a saved campaign that includes the mysterious ASF damage. To see the bug in action, follow the same procedure as above, except in step 3, choose the Sagittarii 2. In step 4, right click and choose "GM Mode > Edit Damage", then manually repair the Sagittarii.

[Nine-Tailed Foxes.zip](https://github.com/MegaMek/mekhq/files/479317/Nine-Tailed.Foxes.zip)

**Comment by crazys1** [comment]
I found that if I remove all of the affected units and then add the units again, the mystery damage goes away. I could save the campaign and load it without a problem. However, after completing a few battles, the mystery damage occurred again, but this time in only 2 units.

I hypothesize that If I were to fix the affected units using "Restore Unit" and then complete a few more battles that eventually even more units would have this problem once again. I think that I had so many bugged units before was because I kept restoring the units, then proceeding to play.

I have not yet narrowed down what seems to be the cause of the issue.
